From 91d6255bcf4b703a12b77020e92d2f28343accfe Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Mon, 7 Feb 2022 09:16:37 -0600
Subject: [PATCH 2/2] spirv-cross: use spirv-cross instead of -c-shared

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 src/d3d11/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/d3d11/meson.build b/src/d3d11/meson.build
index 8cde751..0b30c7e 100644
--- a/src/d3d11/meson.build
+++ b/src/d3d11/meson.build
@@ -5,7 +5,7 @@ d3d11_headers_extra = [ # needed internally
   cc.check_header('dxgi1_6.h', required: d3d11),
 ]
 d3d11_deps = [
-  dependency('spirv-cross-c-shared', version: '>=0.29.0', required: d3d11),
+  dependency('spirv-cross', version: '>=0.29.0', required: d3d11),
   cc.find_library('version', required: d3d11),
 ]
 
-- 
2.37.3

