From aca0fa69e0c8025d0e2741e95225b6fa2ac9f16f Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Mon, 7 Feb 2022 09:16:37 -0600
Subject: [PATCH 2/2] spirv-cross: use spirv-cross instead of -c-shared

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 src/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/meson.build b/src/meson.build
index 57bbd57..c136d40 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -37,7 +37,7 @@ test_deps = []
 vulkan_opt = get_option('vulkan')
 vulkan = dependency('vulkan', version: '>=1.2.0', required: vulkan_opt)
 shaderc = dependency('shaderc_combined', version: '>=2019.1', required: get_option('shaderc'))
-cross = dependency('spirv-cross-c-shared', version: '>=0.29.0', required: get_option('d3d11'))
+cross = dependency('spirv-cross', version: '>=0.29.0', required: get_option('d3d11'))
 
 opengl = disabler()
 if not get_option('opengl').disabled()
-- 
2.37.1

