From d9d0c0ed3c391be667b7f49b52abdaabf9770c3e Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Tue, 20 Oct 2020 22:16:42 +0000
Subject: [PATCH 05/10] include/vlc_common.h: fix snprintf and vsnprintf
 redefined warnings

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 include/vlc_common.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/vlc_common.h b/include/vlc_common.h
index f62ff86d04..6700a12d79 100644
--- a/include/vlc_common.h
+++ b/include/vlc_common.h
@@ -83,8 +83,12 @@
  #define PRIo64 "llo"
  #undef PRIx64
  #define PRIx64 "llx"
+ #ifndef snprintf
  #define snprintf __mingw_snprintf
+ #endif
+ #ifdef vsnprintf
  #define vsnprintf __mingw_vsnprintf
+ #endif
  #define swprintf _snwprintf
 #endif
 
-- 
2.31.1

