From 89730d3ea7da03dc3593cadc4c8b881ba6d52c65 Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Tue, 20 Oct 2020 22:13:27 +0000
Subject: [PATCH 06/12] include/vlc_fixups.h: fix iovec is redefined errors

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 include/vlc_fixups.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/vlc_fixups.h b/include/vlc_fixups.h
index 9149c7519f..6de100a205 100644
--- a/include/vlc_fixups.h
+++ b/include/vlc_fixups.h
@@ -446,11 +446,14 @@ struct timespec {
 #endif
 
 #ifdef _WIN32
+#ifndef STRUCT_IOVEC_DEFINED
+#define STRUCT_IOVEC_DEFINED 1
 struct iovec
 {
     void  *iov_base;
     size_t iov_len;
 };
+#endif
 #define IOV_MAX 255
 struct msghdr
 {
-- 
2.31.1

