From 7c006c732c84ece5999e1a343695e5bb8e6ca668 Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Sat, 13 Mar 2021 16:30:25 -0600
Subject: [PATCH 13/13] medialibrary: add missing parameter for media()

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 modules/misc/medialibrary/medialibrary.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/misc/medialibrary/medialibrary.cpp b/modules/misc/medialibrary/medialibrary.cpp
index 1c8badd151..729dcf375a 100644
--- a/modules/misc/medialibrary/medialibrary.cpp
+++ b/modules/misc/medialibrary/medialibrary.cpp
@@ -1680,7 +1680,7 @@ int MediaLibrary::listPlaylist( int listQuery, const medialibrary::QueryParamete
             if ( pattern != nullptr )
                 query = playlist->searchMedia( pattern, paramsPtr );
             else
-                query = playlist->media();
+                query = playlist->media( paramsPtr );
             if ( query == nullptr )
                 return VLC_EGENERIC;
             switch ( listQuery )
-- 
2.27.0

