From bc92757a078a8ef4c00b9189964a62c616f6d53f Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Sat, 4 Sep 2021 00:18:00 -0500
Subject: [PATCH] jv_thread: try using HAVE_PTHREAD_KEY_CREATE instead

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 src/jv.c        | 2 +-
 src/jv_thread.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/jv.c b/src/jv.c
index 6de4150..268d5ac 100644
--- a/src/jv.c
+++ b/src/jv.c
@@ -221,7 +221,7 @@ enum {
 #define DEC_NUMBER_STRING_GUARD (14)
 
 #include "jv_thread.h"
-#ifdef WIN32
+#if !defined(HAVE_PTHREAD_KEY_CREATE)
 /* Copied from Heimdal: thread-specific keys; see lib/base/dll.c in Heimdal */
 
 /*
diff --git a/src/jv_thread.h b/src/jv_thread.h
index a34cd08..d22434b 100644
--- a/src/jv_thread.h
+++ b/src/jv_thread.h
@@ -1,7 +1,7 @@
 #ifndef JV_THREAD_H
 #define JV_THREAD_H
 
-#ifdef WIN32
+#if !defined(HAVE_PTHREAD_KEY_CREATE)
 #include <windows.h>
 #include <winnt.h>
 #include <errno.h>
-- 
2.41.0

