From 93f146cb1a20d2a92c9eeb84f587545eb24ff8b8 Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Mon, 9 Mar 2020 03:38:24 +0000
Subject: [PATCH 1/2] tiffgt: Link winmm if windows

---
 tools/CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index 66b8e0bb..2eb18ac1 100755
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -129,7 +129,9 @@ if(HAVE_OPENGL)
   target_sources(tiffgt PRIVATE tiffgt.c)
   target_link_libraries(tiffgt PRIVATE tiff port)
   target_link_libraries(tiffgt PRIVATE OpenGL::GL GLUT::GLUT)
-
+  if(WIN32)
+    target_link_libraries(tiffgt winmm)
+  endif()
   install(TARGETS tiffgt
           RUNTIME DESTINATION "${CMAKE_INSTALL_FULL_BINDIR}")
 endif()
-- 
2.25.1

