From b13045d63fd4f5d7ca4d2acf2643bb9692463bfa Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Fri, 11 Sep 2020 05:01:10 +0000
Subject: [PATCH 2/2] CMake: Set CMP0060 to NEW for glut libs

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ac649217..75b40626 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,6 +28,9 @@ cmake_minimum_required(VERSION 3.9.0)
 
 # Default policy is from 3.9.0
 cmake_policy(VERSION 3.9.0)
+if(POLICY CMP0060)
+  cmake_policy(SET CMP0060 NEW)
+endif(POLICY CMP0060)
 
 # Find CMake modules in cmake/
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
-- 
2.25.1

