From 2387c818d917eabe2a2973ca64cdd1a201745195 Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Thu, 13 Mar 2025 19:03:18 -0500
Subject: [PATCH 1/2] ae_lavc: fix deprecated warnings

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 libmpcodecs/ae_lavc.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/libmpcodecs/ae_lavc.c b/libmpcodecs/ae_lavc.c
index a9f49d009d..3a25e8cbee 100644
--- a/libmpcodecs/ae_lavc.c
+++ b/libmpcodecs/ae_lavc.c
@@ -42,7 +42,7 @@
 #include "libmpdemux/mp_taglists.h"
 #include "fmt-conversion.h"
 
-static AVCodec        *lavc_acodec;
+static const AVCodec  *lavc_acodec;
 static AVCodecContext *lavc_actx;
 static int compressed_frame_size = 0;
 
@@ -140,6 +140,7 @@ static int get_frame_size(audio_encoder_t *encoder)
 
 int mpae_init_lavc(audio_encoder_t *encoder)
 {
+	const enum AVSampleFormat *sample_fmts = NULL;
 	encoder->params.samples_per_frame = encoder->params.sample_rate;
 	encoder->params.bitrate = encoder->params.sample_rate * encoder->params.channels * 2 * 8;
 
@@ -177,10 +178,11 @@ int mpae_init_lavc(audio_encoder_t *encoder)
 	lavc_actx->codec_id = lavc_acodec->id;
 	// put sample parameters
 	lavc_actx->sample_fmt = AV_SAMPLE_FMT_S16;
-	if (lavc_acodec->sample_fmts) {
+	avcodec_get_supported_config(lavc_actx, lavc_acodec, AV_CODEC_CONFIG_SAMPLE_FORMAT, 0, (const void **)&sample_fmts, NULL);
+	if (sample_fmts) {
 		const enum AVSampleFormat *fmts;
-		lavc_actx->sample_fmt = lavc_acodec->sample_fmts[0]; // fallback to first format
-		for (fmts = lavc_acodec->sample_fmts; *fmts != AV_SAMPLE_FMT_NONE; fmts++) {
+		lavc_actx->sample_fmt = sample_fmts[0]; // fallback to first format
+		for (fmts = sample_fmts; *fmts != AV_SAMPLE_FMT_NONE; fmts++) {
 			if (samplefmt2affmt(av_get_packed_sample_fmt(*fmts)) == encoder->params.sample_format) { // preferred format found
 				lavc_actx->sample_fmt = *fmts;
 				break;
-- 
2.48.1

