From 64018ceb9fcf92b2e9fd8b6e5f87503612138fbc Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Sat, 21 May 2022 14:34:18 -0500
Subject: [PATCH 4/5] pkgconfig: fix pkg-config file for mingw64

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 etc/luajit.pc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/etc/luajit.pc b/etc/luajit.pc
index 96433008..f2b1aa28 100644
--- a/etc/luajit.pc
+++ b/etc/luajit.pc
@@ -21,5 +21,5 @@ URL: https://luajit.org
 Version: ${version}
 Requires:
 Libs: -L${libdir} -l${libname}
-Libs.private: -Wl,-E -lm -ldl
+Libs.private: -lm -liconv
 Cflags: -I${includedir}
-- 
2.48.1

