From f639c093b07e72a16f24ef0a44407b5e546ef988 Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Fri, 8 Dec 2023 10:19:31 -0600
Subject: [PATCH] ao_wasapi_utils: include mmreg.h for WAVE_FORMAT*

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 audio/out/ao_wasapi_utils.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/audio/out/ao_wasapi_utils.c b/audio/out/ao_wasapi_utils.c
index 97a8e17b2f..eb41eb80be 100644
--- a/audio/out/ao_wasapi_utils.c
+++ b/audio/out/ao_wasapi_utils.c
@@ -21,6 +21,7 @@
 #include <wchar.h>
 #include <windows.h>
 #include <errors.h>
+#include <mmreg.h>
 #include <ksguid.h>
 #include <ksmedia.h>
 #include <avrt.h>
-- 
2.43.0

