From ad6b639a5a945d6ccc7a08b13ac0f0273c2da54a Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Thu, 21 Jan 2021 17:13:54 +0000
Subject: [PATCH 2/2] Findrav1e: fix include path

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 cmake/Modules/Findrav1e.cmake | 22 ++++++++++++++--------
 1 file changed, 14 insertions(+), 8 deletions(-)

diff --git a/cmake/Modules/Findrav1e.cmake b/cmake/Modules/Findrav1e.cmake
index 05cbb94..9cb0f18 100644
--- a/cmake/Modules/Findrav1e.cmake
+++ b/cmake/Modules/Findrav1e.cmake
@@ -19,17 +19,23 @@
 
 find_package(PkgConfig QUIET)
 if (PKG_CONFIG_FOUND)
-    pkg_check_modules(_RAV1E rav1e)
+    pkg_check_modules(RAV1E rav1e)
 endif (PKG_CONFIG_FOUND)
 
-find_path(RAV1E_INCLUDE_DIR
-          NAMES rav1e/rav1e.h
-          PATHS ${_RAV1E_INCLUDEDIR}
-)
+if (NOT RAV1E_INCLUDE_DIRS)
+    find_path(RAV1E_INCLUDE_DIRS
+        NAMES rav1e.h
+        NAMES ${CMAKE_INSTALL_INCLUDEDIR}
+    )
+endif (NOT RAV1E_INCLUDE_DIRS)
 
-find_library(RAV1E_LIBRARY
-             NAMES rav1e
-             PATHS ${_RAV1E_LIBDIR})
+set(RAV1E_INCLUDE_DIR ${RAV1E_INCLUDE_DIRS})
+
+if (NOT RAV1E_LIBRARY)
+    find_library(RAV1E_LIBRARY
+        NAMES rav1e
+        PATHS ${_RAV1E_LIBDIR})
+endif (NOT RAV1E_LIBRARY)
 
 if (RAV1E_LIBRARY)
     set(RAV1E_LIBRARIES
-- 
2.25.1

