From 7ed795ccdb5ad66cf5b5f2d2241c41553721ae5e Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Thu, 24 Sep 2020 03:35:43 +0000
Subject: [PATCH 3/6] Remove wine prefix before fxc2

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 src/gui/configure.pri | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gui/configure.pri b/src/gui/configure.pri
index 9f3ea8c9e3..b113c5c391 100644
--- a/src/gui/configure.pri
+++ b/src/gui/configure.pri
@@ -24,7 +24,7 @@ defineTest(qtConfLibrary_freetype) {
 # DXSDK_DIR variable. Starting with Windows Kit 8, it is included in
 # the Windows SDK.
 defineTest(qtConfTest_fxc) {
-	fxc = wine "$$(DXSDK_DIR)/fxc2.exe"
+	fxc = "$$(DXSDK_DIR)/fxc2.exe"
 
     !isEmpty(fxc) {
         $${1}.value = $$clean_path($$fxc)
-- 
2.25.1

