From 7efef7d8cad310802f1eb66360935d9cf9a2e988 Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Sun, 4 Feb 2024 00:12:36 -0600
Subject: [PATCH 26/27] ShaderVars: include cstdint for uint64_t

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 src/3rdparty/angle/include/GLSLANG/ShaderVars.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/3rdparty/angle/include/GLSLANG/ShaderVars.h b/src/3rdparty/angle/include/GLSLANG/ShaderVars.h
index 709428aebf..cd1de0aae0 100644
--- a/src/3rdparty/angle/include/GLSLANG/ShaderVars.h
+++ b/src/3rdparty/angle/include/GLSLANG/ShaderVars.h
@@ -14,6 +14,7 @@
 #include <array>
 #include <string>
 #include <vector>
+#include <cstdint>
 
 // This type is defined here to simplify ANGLE's integration with glslang for SPIRv.
 using ShCompileOptions = uint64_t;
-- 
2.43.0

