From 6db9eb6f3118b66df1c2881742020c4e91301ebe Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Sat, 22 May 2021 17:52:58 -0500
Subject: [PATCH 1/4] vwebp: Use GLUT and opengl import targets

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index efa93ed9..e821c676 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -514,9 +514,9 @@ if(WEBP_BUILD_VWEBP)
     parse_makefile_am(${CMAKE_CURRENT_SOURCE_DIR}/examples "VWEBP_SRCS" "vwebp")
     add_executable(vwebp ${VWEBP_SRCS})
     target_link_libraries(vwebp
-                          ${OPENGL_LIBRARIES}
+                          GLUT::GLUT
+                          OpenGL::GL
                           exampleutil
-                          ${GLUT_glut_LIBRARY}
                           imageioutil
                           webp
                           webpdemux)
-- 
2.31.1

