From ca3bacacf866383b10785a7eba036d5dbfe80760 Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Wed, 18 Aug 2021 11:02:41 -0500
Subject: [PATCH 2/2] pc: remove CMAKE_CXX_IMPLICIT_LINK_LIBRARIES

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 loader/CMakeLists.txt | 13 +++++--------
 1 file changed, 5 insertions(+), 8 deletions(-)

diff --git a/loader/CMakeLists.txt b/loader/CMakeLists.txt
index 713d05400..2e1a2c06d 100644
--- a/loader/CMakeLists.txt
+++ b/loader/CMakeLists.txt
@@ -380,14 +380,11 @@ endif()
 find_package(PkgConfig QUIET)
 if(PKG_CONFIG_FOUND)
     set(PRIVATE_LIBS "")
-    if (APPLE AND BUILD_STATIC_LOADER)
-        # Libs.private should only be present when building a static loader
-        foreach(LIB ${CMAKE_CXX_IMPLICIT_LINK_LIBRARIES})
-            list(APPEND PRIVATE_LIBS "-l${LIB}")
-        endforeach()
-        list(REMOVE_DUPLICATES PRIVATE_LIBS)
-        set(PRIVATE_LIBS "Libs.private: ${PRIVATE_LIBS}")
-    endif()
+    foreach(LIB ${PLATFORM_LIBS})
+        list(APPEND PRIVATE_LIBS "-l${LIB}")
+    endforeach()
+    list(REMOVE_DUPLICATES PRIVATE_LIBS)
+    set(PRIVATE_LIBS "Libs.private: ${PRIVATE_LIBS}")
     if(WIN32)
         if(MINGW)
             set(VULKAN_LIB_SUFFIX "-1.dll")
-- 
2.39.1

