From cdb3364b413fc4c23a0141e22c96f8a01f6eddc0 Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Thu, 8 May 2025 16:56:50 -0500
Subject: [PATCH 5/6] loader: dllmain related hacks

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 loader/CMakeLists.txt   | 1 +
 loader/loader_windows.c | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/loader/CMakeLists.txt b/loader/CMakeLists.txt
index 6797e86bb..54a30cdef 100644
--- a/loader/CMakeLists.txt
+++ b/loader/CMakeLists.txt
@@ -372,6 +372,7 @@ if(WIN32)
 
     if (BUILD_STATIC_LOADER)
         set(VULKAN_LIBRARY_TYPE STATIC)
+        target_compile_definitions(loader_specific_options INTERFACE BUILD_STATIC_LOADER)
     else()
         set(VULKAN_LIBRARY_TYPE SHARED)
     endif()
diff --git a/loader/loader_windows.c b/loader/loader_windows.c
index 91cdfd00b..64ef9b357 100644
--- a/loader/loader_windows.c
+++ b/loader/loader_windows.c
@@ -95,6 +95,7 @@ void windows_initialization(void) {
 #endif
 }
 
+#if !defined(BUILD_STATIC_LOADER)
 BOOL WINAPI DllMain(HINSTANCE hinst, DWORD reason, LPVOID reserved) {
     (void)hinst;
     switch (reason) {
@@ -116,6 +117,7 @@ BOOL WINAPI DllMain(HINSTANCE hinst, DWORD reason, LPVOID reserved) {
     }
     return TRUE;
 }
+#endif
 
 bool windows_add_json_entry(const struct loader_instance *inst,
                             char **reg_data,    // list of JSON files
-- 
2.52.0

