From ae51b80d475f1f859790d7159f118d8cca282fa6 Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Tue, 9 Jan 2024 16:28:35 -0600
Subject: [PATCH] configure: use pkg-config for psl

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 configure.ac | 28 ++++++++++++++++++++++------
 1 file changed, 22 insertions(+), 6 deletions(-)

diff --git a/configure.ac b/configure.ac
index 99e00514a..06d82e69f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2059,12 +2059,28 @@ AC_ARG_WITH(libpsl,
            with_libpsl=yes)
 curl_psl_msg="no      (libpsl disabled)"
 if test $with_libpsl != "no"; then
-  AC_SEARCH_LIBS(psl_builtin, psl,
-    [curl_psl_msg="enabled";
-     AC_DEFINE([USE_LIBPSL], [1], [PSL support enabled])
-     ],
-    [AC_MSG_ERROR([libpsl was not found]) ]
-  )
+  CURL_CHECK_PKGCONFIG(libpsl)
+  if test "$PKGCONFIG" != "no" ; then
+    LIB_PSL=`$PKGCONFIG --libs-only-l libpsl`
+    LD_PSL=`$PKGCONFIG --libs-only-L libpsl`
+    CPP_PSL=`$PKGCONFIG --cflags-only-I libpsl`
+    version=`$PKGCONFIG --modversion libpsl`
+    DIR_PSL=`echo $LD_PSL | $SED -e 's/-L//'`
+
+    AC_DEFINE([USE_LIBPSL], [1], [PSL support enabled])
+    LDFLAGS="$LDFLAGS $LD_PSL"
+    CPPFLAGS="$CPPFLAGS $CPP_PSL"
+    LIBS="$LIB_PSL $LIBS"
+  else
+    dnl no nghttp2 pkg-config found, deal with it
+    AC_SEARCH_LIBS(psl_builtin, psl,
+      [curl_psl_msg="enabled";
+       AC_DEFINE([USE_LIBPSL], [1], [PSL support enabled])
+       ],
+      [AC_MSG_ERROR([libpsl was not found]) ]
+    )
+  fi
+
 fi
 AM_CONDITIONAL([USE_LIBPSL], [test "$curl_psl_msg" = "enabled"])
 
-- 
2.43.0

